dotcom-rendering/src/components/ShareButton.importable.tsx

```diff
@@ -0,0 +1,229 @@
+	color: ${themePalette('--share-button-copied')};
+	padding: ${sizeXSmall ? '0 4px' : '0 10px'};
+
+	svg {
+		fill: ${palette.success[400]};
+	}
+`;
+
+const buttonStyles = css`
+	color: ${themePalette('--share-button')};
+	svg {
+		fill: ${themePalette('--share-button')};
+	}
+	:hover {
+		background-color: ${themePalette('--share-button')};
+		border-color: ${themePalette('--share-button')};
+		color: ${themePalette('--share-button-hover')};
+		svg {
+			fill: ${themePalette('--share-button-hover')};
+		}
+	}
+`;
+
+const nativeShare = (sizeXSmall: boolean) => css`
+	width: ${sizeXSmall ? '79px' : '105px'};
+	border-color: ${themePalette('--share-button-border')};
+`;
+
+const hoverStyles = css`
+	:hover {
+		color: ${themePalette('--share-button-liveblog-mobile')};
+		background-color: ${palette.neutral[100]};
+		svg {
+			fill: ${themePalette('--share-button-liveblog-mobile')};
+		}
+	}
+`;
+
+const liveBlogMobile = (isCopied: boolean) => css`
+	${until.desktop} {
+		color: ${palette.neutral[100]};
+		border-color: ${transparentColour(palette.neutral[100], 0.4)};
+		svg {
+			fill: ${isCopied ? palette.success[500] : palette.neutral[100]};
+		}
+		${!isCopied && hoverStyles}
+	}
+`;
+
+const getUrl = ({
+	pageId,
+	CMP,
+	blockId,
+}: {
+	pageId: string;
+	CMP?: string;
+	blockId?: string;
+}) => {
+	const searchParams = new URLSearchParams({});
+	if (CMP) searchParams.append('CMP', CMP); // Do we want to track this?
+	if (blockId) searchParams.append('page', `with:block-${blockId}`);
+
+	const blockHash = blockId ? `#block-${blockId}` : '';
+	const paramsString = searchParams.toString();
+	return new URL(
+		`${pageId}${paramsString ? '?' + paramsString : ''}${blockHash}`,
+		'https://www.theguardian.com/',
+	).href;
+};
+
+export const NativeShareButton = ({
+	onShare,
+	size,
+	isLiveBlogArticleMeta,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogArticleMeta: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+
+	return (
+		<Button
+			onClick={onShare}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={<SvgShare />}
+			css={[
+				buttonStyles,
+				nativeShare(sizeXSmall),
+				isLiveBlogArticleMeta && liveBlogMobile(false),
+			]}
+		>
+			Share
+		</Button>
+	);
+};
+
+export const CopyLinkButton = ({
+	onShare,
+	size,
+	isLiveBlogArticleMeta,
+	isCopied,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogArticleMeta: boolean;
+	isCopied: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+	return (
+		<Button
+			onClick={onShare}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={isCopied ? <SvgCheckmark /> : <LinkIcon />}
+			css={[
+				...(isCopied
+					? [copiedButtonStyles(sizeXSmall)]
+					: [buttonStyles]),
+				sharedButtonStyles(sizeXSmall, isCopied),
+				isLiveBlogArticleMeta && liveBlogMobile(isCopied),
+			]}
+		>
+			{isCopied ? 'Link copied' : 'Copy link'}
+		</Button>
+	);
+};
+
+export const ShareButton = ({
```

Comment:
Ideally, this would have a little description, [like other islands](https://assets.guim.co.uk/assets/stats/islands.html), and be added to the `Island.test.tsx` file to demonstrate that it works on the server.